### PR TITLE
[docs] Move all /latest links to /master

### PR DESCRIPTION
**Pull request by ericl**
## Why are these changes needed?

Since latest is a hidden doc version, move them to master. Remove the banner warning from master but keep it for latest (telling users to go to /master).

### CODE REVIEW

File: doc/examples/lm/lm-cluster.yaml [hunk 1]
```diff
@@ -67,7 +67,7 @@ head_node:
 # Provider-specific config for worker nodes, e.g. instance type. By default
 # Ray will auto-configure unspecified fields such as SubnetId and KeyName.
 # For more documentation on available fields, see:
-# http://boto3.readthedocs.io/en/latest/reference/services/ec2.html#EC2.ServiceResource.create_instances
+# http://boto3.readthedocs.io/en/master/reference/services/ec2.html#EC2.ServiceResource.create_instances
```
Comment (richardliaw) [inline doc/examples/lm/lm-cluster.yaml:70]: keep

File: doc/examples/lm/lm-cluster.yaml [hunk 2]
```diff
@@ -50,7 +50,7 @@ auth:
 # Provider-specific config for the head node, e.g. instance type. By default
 # Ray will auto-configure unspecified fields such as SubnetId and KeyName.
 # For more documentation on available fields, see:
-# http://boto3.readthedocs.io/en/latest/reference/services/ec2.html#EC2.ServiceResource.create_instances
+# http://boto3.readthedocs.io/en/master/reference/services/ec2.html#EC2.ServiceResource.create_instances
```
Comment (richardliaw) [inline doc/examples/lm/lm-cluster.yaml:53]: keep

File: doc/examples/newsreader/ray-project/project.yaml [hunk 3]
```diff
@@ -4,7 +4,7 @@ name: ray-example-newsreader
 
 description: "A simple news reader example that uses ray actors to serve requests"
 tags: ["ray-example", "flask", "rss", "newsreader"]
-documentation: https://docs.ray.io/en/latest/auto_examples/plot_newsreader.html
+documentation: https://docs.ray.io/en/master/auto_examples/plot_newsreader.html
```
Comment (richardliaw) [inline doc/examples/newsreader/ray-project/project.yaml:7]: we should just get rid of all of these ray-project/* things

File: release/stress_tests/cluster.yaml [hunk 4]
```diff
@@ -41,7 +41,7 @@ auth:
 # Provider-specific config for the head node, e.g. instance type. By default
 # Ray will auto-configure unspecified fields such as SubnetId and KeyName.
 # For more documentation on available fields, see:
-# http://boto3.readthedocs.io/en/latest/reference/services/ec2.html#EC2.ServiceResource.create_instances
+# http://boto3.readthedocs.io/en/master/reference/services/ec2.html#EC2.ServiceResource.create_instances
```
Comment (richardliaw) [inline release/stress_tests/cluster.yaml:44]: revert all boto links

File: release/rllib_tests/stress_tests/cluster.yaml [hunk 5]
```diff
@@ -41,7 +41,7 @@ auth:
 # Provider-specific config for the head node, e.g. instance type. By default
```
Comment (richardliaw) [inline release/rllib_tests/stress_tests/cluster.yaml:41]: revert all boto links

File: python/ray/util/sgd/torch/ptl_operator.py [hunk 6]
```diff
@@ -62,7 +62,7 @@ def scheduler_dicts(self):
         returns scheduler objects.
 
         See
-        https://pytorch-lightning.readthedocs.io/en/latest/lightning_module.html#configure-optimizers
+        https://pytorch-lightning.readthedocs.io/en/master/lightning_module.html#configure-optimizers
```
Comment (richardliaw) [inline python/ray/util/sgd/torch/ptl_operator.py:65]: revert

File: python/ray/serve/http_proxy.py [hunk 7]
```diff
@@ -75,7 +75,7 @@ async def _handle_system_request(self, scope, receive, send):
 
     async def __call__(self, scope, receive, send):
         # NOTE: This implements ASGI protocol specified in
-        #       https://asgi.readthedocs.io/en/latest/specs/index.html
+        #       https://asgi.readthedocs.io/en/master/specs/index.html
```
Comment (richardliaw) [inline python/ray/serve/http_proxy.py:78]: revert

File: python/ray/autoscaler/aws/example-ml.yaml [hunk 8]
```diff
@@ -73,7 +73,7 @@ auth:
 # Provider-specific config for the head node, e.g. instance type. By default
 # Ray will auto-configure unspecified fields such as SubnetId and KeyName.
 # For more documentation on available fields, see:
-# http://boto3.readthedocs.io/en/latest/reference/services/ec2.html#EC2.ServiceResource.create_instances
+# http://boto3.readthedocs.io/en/master/reference/services/ec2.html#EC2.ServiceResource.create_instances
```
Comment (richardliaw) [inline python/ray/autoscaler/aws/example-ml.yaml:76]: revert

File: doc/source/tune/contrib.rst [hunk 9]
```diff
@@ -57,7 +57,7 @@ burden and speedup review process.
 
     pytest ray/python/ray/tune/tests/
 
-Documentation should be documented in `Google style <https://sphinxcontrib-napoleon.readthedocs.io/en/latest/example_google.html>`__ format.
+Documentation should be documented in `Google style <https://sphinxcontrib-napoleon.readthedocs.io/en/master/example_google.html>`__ format.
```
Comment (richardliaw) [inline doc/source/tune/contrib.rst:60]: revert

File: doc/source/tune/_tutorials/tune-xgboost.rst [hunk 10]
```diff
@@ -485,5 +485,5 @@ Further References
 ------------------
 
 * `XGBoost Hyperparameter Tuning - A Visual Guide <https://kevinvecmanis.io/machine%20learning/hyperparameter%20tuning/dataviz/python/2019/05/11/XGBoost-Tuning-Visual-Guide.html>`_
-* `Notes on XGBoost Parameter Tuning <https://xgboost.readthedocs.io/en/latest/tutorials/param_tuning.html>`_
+* `Notes on XGBoost Parameter Tuning <https://xgboost.readthedocs.io/en/master/tutorials/param_tuning.html>`_
```
Comment (richardliaw) [inline doc/source/tune/_tutorials/tune-xgboost.rst:488]: revert

File: doc/source/tune/_tutorials/tune-xgboost.rst [hunk 11]
```diff
@@ -121,7 +121,7 @@ is a ``dict`` in which you can specify parameters for the XGBoost algorithm. In
 simple example, the only parameters we passed are the ``objective`` and ``eval_metric`` parameters.
 The value ``binary:logistic`` tells XGBoost that we aim to train a logistic regression model for
 a binary classification task. You can find an overview over all valid objectives
-`here in the XGBoost documentation <https://xgboost.readthedocs.io/en/latest/parameter.html#learning-task-parameters>`_.
+`here in the XGBoost documentation <https://xgboost.readthedocs.io/en/master/parameter.html#learning-task-parameters>`_.
```
Comment (richardliaw) [inline doc/source/tune/_tutorials/tune-xgboost.rst:124]: revert

**Review by richardliaw — CHANGES_REQUESTED** [review]
Please revert all links that are non-ray related

**Review by ericl — COMMENTED** [review]
Fixed

**Comment by ericl** [comment]
Ping @richardliaw

**Review by richardliaw — APPROVED** [review]

**Pull request merged**